plugins/inputs/nats/nats.go

```diff
@@ -0,0 +1,104 @@
+package nats
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"time"
+
+	"encoding/json"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+
+	gnatsd "github.com/nats-io/gnatsd/server"
+)
+
+type Nats struct {
+	Server          string
+	ResponseTimeout internal.Duration
+}
+
+var sampleConfig = `
+  ## The address of the monitoring endpoint of the NATS server
+  server = "http://localhost:1337"
+
+  ## Maximum time to receive response
+  # response_timeout = "5s"
+`
+
+func (n *Nats) SampleConfig() string {
+	return sampleConfig
+}
+
+func (n *Nats) Description() string {
+	return "Provides metrics about the state of a NATS server"
+}
+
+func (n *Nats) Gather(acc telegraf.Accumulator) error {
+	url, err := url.Parse(n.Server)
+	if err != nil {
+		return err
+	}
+	url.Path = path.Join(url.Path, "varz")
+
+	client := n.createHTTPClient()
+	resp, err := client.Get(url.String())
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	bytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+
+	stats := new(gnatsd.Varz)
+	err = json.Unmarshal([]byte(bytes), &stats)
+	if err != nil {
+		return err
+	}
+
+	acc.AddFields("nats_varz",
```

Comment:
What do you think about just calling this nats?  Adding varz could be useful if we anticipate other measurements but is otherwise unneeded IMO.

Reply:
Ha! It was called `nats` originally but on #3186 you suggested using `nats_vars` :)

Some of the other measurements that NATS offers through other monitoring endpoints could be useful to expose at some stage (e.g detailed connection & subscription metrics) but right now it's just the high level metrics that are being exposed and there's no need to include "varz" in the name for those. 

I'm happy to change it back to just `nats`. We can add suffixes for the more detailed stuff.

Reply:
:/ I guess this probably means it doesn't matter much, lets just go with what we have.